numpy/core/tests/test_datetime.py

```diff
@@ -75,6 +75,15 @@ def test_datetime_casting_rules(self):
         # Can cast safely/same_kind from integer to timedelta
         assert_(np.can_cast('i8', 'm8', casting='same_kind'))
         assert_(np.can_cast('i8', 'm8', casting='safe'))
+        assert_(np.can_cast('i4', 'm8', casting='same_kind'))
+        assert_(np.can_cast('i4', 'm8', casting='safe'))
+        assert_(np.can_cast('u4', 'm8', casting='same_kind'))
+        assert_(np.can_cast('u4', 'm8', casting='safe'))
+
+        # Cannot cast safely from unsigned integer of the same size, which
+        # could overflow
+        assert_(np.can_cast('u8', 'm8', casting='same_kind'))
+        assert_(not np.can_cast('u8', 'm8', casting='safe'))
```

Comment:
Yes, these tests are very clean/nice. Seems pretty clear that this is type-size, not value-size.